commit 374fbed3da0197f794a904e78e75e961c7e2e92c
Author: Mark Wielaard <mark@klomp.org>
Date:   Wed Dec 4 00:39:26 2019 +0100

    debuginfod: Fix implicit conversion from 'CURLcode' to 'CURLMcode'
    
    GCC10 warns when converting the value of one enum type into another:
    
    debuginfod-client.c:530:24: error: implicit conversion from ‘CURLcode’
                                       to ‘CURLMcode’ [-Werror=enum-conversion]
      530 |               curl_res = curl_easy_getinfo(target_handle,
          |                        ^
    
    libcurl has different error code enums. The "easy" interfaces return
    a CURLcode error. The "multi" interface functions return a CURLMcode.
    
    Signed-off-by: Mark Wielaard <mark@klomp.org>

diff --git a/debuginfod/debuginfod-client.c b/debuginfod/debuginfod-client.c
index 6e62b86c..302ea2dc 100644
--- a/debuginfod/debuginfod-client.c
+++ b/debuginfod/debuginfod-client.c
@@ -509,8 +509,6 @@ debuginfod_query_server (debuginfod_client *c,
   long loops = 0;
   do
     {
-      CURLMcode curl_res;
-
       if (c->progressfn) /* inform/check progress callback */
         {
           loops ++;
@@ -518,6 +516,7 @@ debuginfod_query_server (debuginfod_client *c,
           long pb = 0;
           if (target_handle) /* we've committed to a server; report its download progress */
             {
+              CURLcode curl_res;
 #ifdef CURLINFO_SIZE_DOWNLOAD_T
               curl_off_t dl;
               curl_res = curl_easy_getinfo(target_handle,
@@ -564,10 +563,10 @@ debuginfod_query_server (debuginfod_client *c,
           if (data[i].handle != target_handle)
             curl_multi_remove_handle(curlm, data[i].handle);
 
-      curl_res = curl_multi_perform(curlm, &still_running);
-      if (curl_res != CURLM_OK)
+      CURLMcode curlm_res = curl_multi_perform(curlm, &still_running);
+      if (curlm_res != CURLM_OK)
         {
-          switch (curl_res)
+          switch (curlm_res)
             {
             case CURLM_CALL_MULTI_PERFORM: continue;
             case CURLM_OUT_OF_MEMORY: rc = -ENOMEM; break;
